here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/tasks/StorageApiTask.java

```diff
@@ -149,4 +155,22 @@ private static String mismatchMsg(String storageIdFromPath, Storage storageFromB
     return "Mismatch between storage ids. Path storage id: %s, body storage id: %s"
         .formatted(storageIdFromPath, storageFromBody.getId());
   }
+
+  private static final String JSON_KEY_PASSWORD = "password";
+
+  private void removePasswordFromProps(Map<String, Object> propertiesAsMap) {
+    for (Entry<String, Object> entry : propertiesAsMap.entrySet()) {
+      if (Objects.equals(entry.getKey(), JSON_KEY_PASSWORD)) {
+        entry.setValue("xxxxxx");
+      } else if (entry.getValue() instanceof Map) {
+        // recursive call to the nested json property
+        removePasswordFromProps((Map<String, Object>) entry.getValue());
+      } else if (entry.getValue() instanceof ArrayList array) {
+        // recursive call to the nested array json
+        for (Object arrayEntry : array) {
+          removePasswordFromProps((Map<String, Object>) arrayEntry);
+        }
+      }
+    }
+  }
```

Comment:
I think this code would look nice in dedicated utility class - something "password specific" like `PasswordObscurityUtil` for example. Then `StorageApi` would only delegate this work there and minimize the scope of the Api itself. 
This would be also pretty cool from testing perspective - you would have a sleak way to write a unit test without needing to make a rest call (it's fine that you test it in REST scenarios as well, but dedicated UT seems sensible for such "mechanical" tasks

Reply:
liked this idea. although small change but valuable.